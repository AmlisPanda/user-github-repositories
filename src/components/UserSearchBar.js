import React from 'react';

export class UserSearchBar extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            username: ""
        }

        this.usernameChange = this.usernameChange.bind(this);
    }

    usernameChange(event) {
        this.setState({
            username: event.target.value
        })
    }

    render() {
        return (
            <form className="reposSearch" onSubmit={(e) => this.props.onSearch(e, this.state.username)}>
                <label htmlFor="githubUsername">Github username</label>
                <input type="text" name="githubUsername" value={this.state.username} onChange={this.usernameChange} placeholder="Enter username here" />

                <button>GO</button>
            </form>
        )
    }
}
