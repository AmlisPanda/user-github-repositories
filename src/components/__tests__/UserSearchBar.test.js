import React from 'react';
import renderer from 'react-test-renderer';
import { UserSearchBar } from '../UserSearchBar';

describe('UserSearchBar', () => {
    it('renders the component', () => {
        const tree = renderer.create(<UserSearchBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });

})
