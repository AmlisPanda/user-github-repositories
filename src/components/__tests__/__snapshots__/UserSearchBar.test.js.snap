// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearchBar renders the component 1`] = `
<form
  className="reposSearch"
  onSubmit={[Function]}
>
  <label
    htmlFor="githubUsername"
  >
    Github username
  </label>
  <input
    name="githubUsername"
    onChange={[Function]}
    placeholder="Enter username here"
    type="text"
    value=""
  />
  <button>
    GO
  </button>
</form>
`;
